MMTk PR broke Makefile target used to retrieve all deps

We fetch the tarball with all dependencies' sources in our build.

The MMTk PR apparently broke that.

Running `make -C deps getall` on a x86_64 Linux machine gives me:

```
...
make: *** No rule to make target 'get-mmtk_julia', needed by 'getall'.  Stop.
make: Leaving directory '/home/ubuntu/julia-RAI/deps'
```

https://github.com/JuliaLang/julia/blob/d3964b600a5e596bd47cf04b3da4a1e78d4aecaf/deps/mmtk_julia.mk#L27

```sh
$ make -j -C deps getall
make: Entering directory '/mnt/d/jl/julia/deps'
make: *** No rule to make target 'get-mmtk_julia', needed by 'getall'.  Stop.
make: Leaving directory '/mnt/d/jl/julia/deps'
$ make -C deps get-mmtk_julia
make: Entering directory '/mnt/d/jl/julia/deps'
make: *** No rule to make target 'get-mmtk_julia'.  Stop.
make: Leaving directory '/mnt/d/jl/julia/deps'
$ make -C deps print-MMTK_JULIA_SRC_FILE
make: Entering directory '/mnt/d/jl/julia/deps'
MMTK_JULIA_SRC_FILE=
make: Leaving directory '/mnt/d/jl/julia/deps'
```

So `MMTK_JULIA_SRC_FILE` is not defined.

--- a/deps/mmtk_julia.mk
+++ b/deps/mmtk_julia.mk
@@ -6,9 +6,11 @@
 MMTK_MOVING := 0
 MMTK_VARS := MMTK_PLAN=$(MMTK_PLAN) MMTK_MOVING=$(MMTK_MOVING)
 
+$(eval $(call git-external,mmtk_julia,MMTK_JULIA,,,$(BUILDDIR)))
+get-mmtk_julia: $(MMTK_JULIA_SRC_FILE)
+
 # Download the binding, build it from source
 ifeq (${MMTK_JULIA_DIR},$(BUILDROOT)/usr/lib/mmtk_julia)
-$(eval $(call git-external,mmtk_julia,MMTK_JULIA,,,$(BUILDDIR)))
 
 MMTK_JULIA_DIR=$(BUILDROOT)/deps/$(BUILDDIR)/$(MMTK_JULIA_SRC_DIR)
 MMTK_JULIA_LIB_PATH=$(MMTK_JULIA_DIR)/mmtk/target/$(MMTK_BUILD)
@@ -24,7 +26,6 @@ $(BUILDROOT)/usr/lib/libmmtk_julia.so: $(MMTK_JULIA_LIB_PATH)/libmmtk_julia.so
 $(MMTK_JULIA_LIB_PATH)/libmmtk_julia.so: $(BUILDDIR)/$(MMTK_JULIA_SRC_DIR)/source-extracted
 	@$(PROJECT_DIRS) $(MMTK_VARS) $(MAKE) -C $(MMTK_JULIA_DIR) $(MMTK_BUILD)
 
-get-mmtk_julia: $(MMTK_JULIA_SRC_FILE)
 extract-mmtk_julia: $(BUILDDIR)/$(MMTK_JULIA_SRC_DIR)/source-extracted
 configure-mmtk_julia: extract-mmtk_julia
 compile-mmtk_julia: $(BUILDROOT)/usr/lib/libmmtk_julia.so

